src/js/script.js

```diff
@@ -369,38 +371,77 @@
       thisCart.dom.wrapper = element;
       thisCart.dom.toggleTrigger = thisCart.dom.wrapper.querySelector(select.cart.toggleTrigger);
       thisCart.dom.productList = thisCart.dom.wrapper.querySelector(select.cart.productList);
+      thisCart.dom.deliveryFee = thisCart.dom.wrapper.querySelector(select.cart.deliveryFee);
+      thisCart.dom.subtotalPrice = thisCart.dom.wrapper.querySelector(select.cart.subtotalPrice);
+      thisCart.dom.totalPrice = thisCart.dom.wrapper.querySelector(select.cart.totalPrice);
+      thisCart.dom.totalNumber = thisCart.dom.wrapper.querySelector(select.cart.totalNumber);
     }
 
     initActions(){
       const thisCart = this;
       thisCart.dom.toggleTrigger.addEventListener('click', function(){
         thisCart.dom.wrapper.classList.toggle(classNames.cart.wrapperActive);
       });
-
+      thisCart.dom.productList.addEventListener('updated', function() {
+        thisCart.update();
+      });
     }
 
     add(menuProduct){
       const thisCart = this;
-      console.log('adding product', menuProduct);
+      //console.log('adding product', menuProduct);
   
       /*generare HTML based on template */
       const generateHTML = templates.cartProduct(menuProduct);
   
       /*create element using utils.createElementFromHTML */
-      menuProduct.element = utils.createDOMFromHTML(generateHTML);
+      const generatedDOM = utils.createDOMFromHTML(generatedHTML);
    
       /*add elemnt to menu */
       thisCart.dom.productList.appendChild(menuProduct.element);
 
       thisCart.products.push(new CartProduct(menuProduct, generatedDOM));
-      console.log('thisCart.products', thisCart.products);
+      //console.log('thisCart.products', thisCart.products);
+      }
+
+    update() {
+      const thisCart = this;
+      
+      thisCart.deliveryFee = settings.cart.defaultDeliveryFee;
+      thisCart.totalNumber = 0;
+      thisCart.subtotalPrice = 0;
+      for (let cartProduct of thisCart.products){
+        thisCart.totalNumber += CartProduct.amount;
+        thisCart.subtotalPrice += CartProduct.price;
```

Comment:
5. Zwróć uwagę, że zmienna, jaką utworzyłaś do iteracji w pętli to `cartProduct`. Tymczasem odwołujesz się do zmiennej pisanej wielką literą `CartProduct`, czyli do klasy.  Powinno to wyglądać tak jak poniżej:
```suggestion
        thisCart.totalNumber += cartProduct.amount;
        thisCart.subtotalPrice += cartProduct.price;
```